M2/Macaulay2/packages/IncidenceCorrespondenceCohomology.m2

```diff
@@ -0,0 +1,2105 @@
+newPackage(
+	"IncidenceCorrespondenceCohomology",
+    	Version => "0.1", 
+    	Date => "February 2, 2024",
+    	Authors => {
+	    {Name => "Annet Kyomuhangi", 
+		  Email => "annet.kyomuhangi@gmail.com", 
+	HomePage=>"https://people.busitema.ac.ug/staff/academic-staff/annet-kyomuhangi"}, 
+	    {Name => "Emanuela Marangone", 
+		  Email => "emanuela.marangone@umanitoba.ca", 
+		  HomePage => "https://emanuelamarangone.wixsite.com/marangone"}, 
+	    {Name => "Claudiu Raicu", 
+		  Email => "craicu@nd.edu", 
+		  HomePage => "https://www3.nd.edu/~craicu/"}, 
+	    {Name => "Ethan Reed",
+		  Email => "ereed4@nd.edu", 
+		  HomePage => "https://sites.google.com/view/ethan-reed/home"}
+	    },
+    	Headline => "Cohomology on the incidence corresponce, bundle of principal parts, and lefschetz properties for monomial complete intersections",
+	Keywords => {"Character formulas", "cohomology of line bundles", "incidence correspondence"," Lefschetz properties", "principal parts", "Han--Monsky representation ring"}
+	--Certification => {
+	  --   "journal name" => "",
+	    -- "journal URI" => "",
+	     --"article title" => "",
+	    -- "acceptance date" => "",
+	    -- "published article URI" => "",
+	    -- "published article DOI" => "",
+	    -- "published code URI" => "",
+	    -- "repository code URI" => "",
+	    -- "release at publication" => "",	    
+	    -- "version at publication" => "",
+	    -- "volume number" => "",
+	    -- "volume URI" => ""
+	     --}
+    	)
+
+
+export {--option 
+"UseConjecture",
+"FindCharacter",
+"Multidegree",
+"GorensteinAlg",
+"MonomialAlg",
+--methods and functions
+"splittingFdr",
+"splittingPrincipalParts", 
+"hanMonsky",
+"hasWLP",
+"monomialCIsWithoutWLP",
+"hasSLP",
+"recursiveDividedCohomology",
+"nimDividedCohomology", 
+"incidenceCohomology"}
+
+
+---------------------------------------------------------------------------------------------------------------------------------------------------
+---------------------------------------------------------------------------------------------------------------------------------------------------
+--Torus equivariant splitting of duals of bundles of principal parts on P^1
+---------------------------------------------------------------------------------------------------------------------------------------------------
+---------------------------------------------------------------------------------------------------------------------------------------------------
+
+
+--Input:
+--prime p and integer n
+--Output:
+--smallest integer e such that n < p^e
+
+boundq = (p, n) ->(
+    counter := 0; q := 1;
+    while n >= q do (counter = counter + 1; q = p*q);
+    return counter
```

Comment:
It's not necessary to use `return` on the last line of a function.  (See https://github.com/Macaulay2/M2/wiki/Package-Writing-Style-Guide for this and lots of other stylistic pointers for packages.)